### ISSUE TITLE
Update home IdP discovery to use new SPI

### ISSUE THREAD

**Issue by xgp** [p2-inc/keycloak-orgs#235]
https://github.com/sventorben/keycloak-home-idp-discovery/pull/346

**Comment by rtufisi** [comment]
New Home IdP Discovery 

![Screenshot from 2024-07-20 08-36-33](https://github.com/user-attachments/assets/402e9e6e-3026-4c3f-91c2-f9fbb8c1fcd1)
![Screenshot from 2024-07-20 08-36-50](https://github.com/user-attachments/assets/12834681-1db0-4228-91f4-3e09215ae5a6)
![Screenshot from 2024-07-20 08-36-59](https://github.com/user-attachments/assets/13622b1b-3fca-4427-b756-0c16cc027ed0)
![Screenshot from 2024-07-20 08-37-06](https://github.com/user-attachments/assets/9f50cc50-65ec-4e4d-956c-6e5a7c6cf102)

**Comment by rtufisi** [comment]
@xgp I was looking at the old form of the Home IdP Discovery config.

![Screenshot from 2024-07-20 08-46-29](https://github.com/user-attachments/assets/6e4d6271-784c-4683-a327-28bff61e8336)


Is the 'Required a Verified Email'  and 'Required a Verified Domain'  still required? Are these our implementations?

**Comment by rtufisi** [comment]
I believe this is the new 'verifiedEmail'
![Screenshot from 2024-07-20 08-51-42](https://github.com/user-attachments/assets/236ef565-2603-4dc7-b5db-8a44e3f12396)

**Comment by xgp** [comment]
@rtufisi 

> Is the 'Required a Verified Email' and 'Required a Verified Domain' still required? Are these our implementations?

Yes. Those are for our implementations. We will still need *'Required a Verified Domain'*

> I believe this is the new 'verifiedEmail'

It looks like he added this to the default.

**Comment by rtufisi** [comment]
Could you please give me a example on how to use the IdpSelectorAuthenticator?

**Comment by rtufisi** [comment]
> @rtufisi
> 
> > Is the 'Required a Verified Email' and 'Required a Verified Domain' still required? Are these our implementations?
> 
> Yes. Those are for our implementations. We will still need _'Required a Verified Domain'_ 
Done
> 
> > I believe this is the new 'verifiedEmail'
> 
> It looks like he added this to the default.

It seems to me that our implementation flag "verifiedEmail" and the source flag "forwardUserWithUnverifiedEmail" have opposite logic @xpg. Does this impact our logic? How could we proceed without affecting the clients?

**Comment by xgp** [comment]
I think we would have to run a "migration" in a `postInit` method to load all existing instances of our Authenticator and migrate their configurations.

**Comment by xgp** [comment]
Pulling the home idp extension is currently blocked by it not being in Maven Central. See https://github.com/sventorben/keycloak-home-idp-discovery/issues/400

Going to try pulling the library in locally with the Maven `system` scope:
```xml
<dependency>
    <groupId>com.sample</groupId>
    <artifactId>sample</artifactId>
    <version>1.0</version>
    <scope>system</scope>
    <systemPath>${project.basedir}/src/main/resources/Name_Your_JAR.jar</systemPath>
</dependency>
```

Interim plan if this ^^^ works:
1. Import the compatible home idp jar into a `libs/` directory in the keycloak-orgs repo
2. In our distribution/docker image `phasetwo-containers` repo, import it into the `libs/ext/` dir
3. Document in the `README` that it is necessary to include the home idp jar
4. Eventually PR his repo with an example of how to release to maven central

**Comment by rtufisi** [comment]
Hei @xgp . I've created a first example of how we could use the  home idp extension as a library. There are several things that block us from using it as a library but I've created a PR on how to use it.

Please check: https://github.com/p2-inc/keycloak-orgs/pull/259/files#diff-ca14b1065de1812332188294559f0f7398a15bb862c910f8d94fb83b1a6ce41b

In order to start the project locally use the following config:
![Screenshot from 2024-07-27 14-38-42](https://github.com/user-attachments/assets/57a0d01a-5dd9-4b3f-a399-fd62a7464aff)

**Comment by rtufisi** [comment]
@xgp from my point of view to be able to fully implement a authenticator using "home IdP discovery"  as a library we would need the following:

- AbstractHomeIdpDiscoveryAuthenticatorFactory to allow a custom authenticator logic
```
    public final Authenticator create(KeycloakSession session) {
        return new HomeIdpDiscoveryAuthenticator(this.discovererConfig);
    }
```

In this way we could implement our custom Authenticator behaviour

- Some classes exposed as public HomeIdpForwarderConfigProperties, HomeIdpAuthenticationFlowContext  etc.

**Comment by rtufisi** [comment]
done
